pr_insight/algo/utils.py

```
@@ -316,47 +316,98 @@ def extract_relevant_lines_str(end_line, files, relevant_file, start_line, deden
+        for ticket_analysis in value:
+            try:
+                ticket_url = ticket_analysis.get('ticket_url', '').strip()
+                explanation = ''
+                ticket_compliance_level = ''  # Individual ticket compliance
+                fully_compliant_str = ticket_analysis.get('fully_compliant_requirements', '').strip()
+                not_compliant_str = ticket_analysis.get('not_compliant_requirements', '').strip()
+                requires_further_human_verification = ticket_analysis.get('requires_further_human_verification',
+                                                                          '').strip()
+
+                if not fully_compliant_str and not not_compliant_str:
+                    get_logger().debug(f"Ticket compliance has no requirements",
```

Ruff (0.8.2)

f-string without any placeholders

(F541)